libs/net/nodenetworkconfigurationpolicy.py

```diff
@@ -0,0 +1,246 @@
+            err_msg += f"{nnce_prefix}: {libnmstate_err}"
+
+        return err_msg
+
+    def _get_failed_nnce(self) -> NodeNetworkConfigurationEnactment:
+        for nnce in self.nnces:
+            try:
+                nnce.wait_for_conditions()
+            except TimeoutExpiredError:
+                self.logger.error(f"Failed to get NNCE {nnce.name} status")
+                continue
+
+            for nnce_cond in nnce.instance.status.conditions:
+                if nnce_cond.type == "Failing" and nnce_cond.status == Resource.Condition.Status.TRUE:
+                    yield nnce
+
+    def _process_failed_status(self, failed_condition_reason: str) -> None:
+        last_err_msg = None
+        for failed_nnce in self._get_failed_nnce():
+            nnce_name = failed_nnce.instance.metadata.name
+            nnce_dict = failed_nnce.instance.to_dict()
+            for cond in nnce_dict["status"]["conditions"]:
+                err_msg = self._get_nnce_error_msg(nnce_name=nnce_name, nnce_condition=cond)
+                if err_msg:
+                    last_err_msg = err_msg
+
+        raise NNCPConfigurationFailed(f"Reason: {failed_condition_reason}\n{last_err_msg}")
+
+    def wait_for_status_success(self) -> list[dict]:
+        failed_condition_reason = self.Conditions.Reason.FAILED_TO_CONFIGURE
+        no_match_node_condition_reason = self.Conditions.Reason.NO_MATCHING_NODE
+
+        samples = TimeoutSampler(
+            wait_timeout=TIMEOUT_90SEC,
+            sleep=TIMEOUT_5SEC,
+            func=lambda: self.instance.status.conditions,
+        )
+        for sample in samples:
+            if sample:
+                for nncp_condition in sample:
+                    if (
+                        nncp_condition["reason"] == self.Conditions.Reason.SUCCESSFULLY_CONFIGURED
+                        and nncp_condition["status"] == self.Condition.Status.TRUE
+                    ):
+                        self.logger.info(f"NNCP {self.name} configured Successfully")
+                        return sample
+
+                    elif nncp_condition["reason"] == no_match_node_condition_reason:
+                        raise NNCPConfigurationFailed(f"{self.name}. Reason: {no_match_node_condition_reason}")
+
+                    elif nncp_condition["reason"] == failed_condition_reason:
+                        self._process_failed_status(failed_condition_reason=failed_condition_reason)
+
+        raise NNCPConfigurationFailed(f"{self.name}. None of the conditions were met.")
+
+    def _absent_interface(self) -> None:
+        desired_state = {}
+        if self._desired_state.interfaces:
+            for _iface in self.res["spec"]["desiredState"]["interfaces"]:
+                _iface["state"] = Resource.Interface.State.ABSENT
+
+            desired_state["interfaces"] = self.res["spec"]["desiredState"]["interfaces"]
+
+        ResourceEditor(patches={self: {"spec": {"desiredState": desired_state}}}).update()
```

Comment:
1. "absent" is not a verb. Perhaps "remove" or "delete": `_delete_interfaces`.
2. You should use the new data structure, not the raw dict. When you patch, then convert it to a dict.
3. Use dict comprehension to build the new data.